### PR TITLE
Local run

### LINKED ISSUE #2: Provide local run option
- Spin flink stand-alone cluster
- Deploy simple job

### PR DESCRIPTION
**Pull request by nielsdenissen**
resolve #2

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1 @@
+flink-sample-job/
```
Comment (mrooding) [inline .dockerignore]: can you add a new line at the end of the file? 🙏

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,6 @@
 cover.out
 cli
+flink-sample-job/target/
```
Comment (mrooding) [inline .gitignore]: these 3 lines can be replaced by one line containing target/

File: README.md [hunk 3]
```diff
@@ -16,6 +16,37 @@ Currently, it supports several features:
 
 For a full overview of the commands and flags, run `flink-job-deployer help`
 
+## How to run locally
+
+To be able to test the deployer locally, follow these steps:
+
+1. `go build ./cmd/cli` (build the CLI tool)
+2. `docker-compose build deployer` (build deployer docker container)
```
Comment (mrooding) [inline README.md]: I wonder if you need to include this step. `docker-compose run deployer help` will build it anyway, right?

File: README.md [hunk 4]
```diff
@@ -16,6 +16,37 @@ Currently, it supports several features:
...
+5. `docker-compose run deployer help` (run the Flink deployer with argument `help`)
+
+Repeat step 4 with any commands you'd like to try. 
+
+### Run a sample job
+Provided you ran step 3 of the above guide, a jar with a sample Flink job is available in the deployer. It will be mounted in the deployer container at the following path:
+
+    /tmp/flink-sample-job/flink-stateful-wordcount-assembly-0.jar
+
+To deploy it you can simply run (it's the default command specified in the `docker-compose.yml`):
```
Comment (mrooding) [inline README.md]: The sentence between the parenthesis doesn't read very well to me. I think it should be fine to omit it entirely

File: README.md [hunk 5]
```diff
@@ -16,6 +16,37 @@ Currently, it supports several features:
...
+
+    /tmp/flink-sample-job/flink-stateful-wordcount-assembly-0.jar
+
+To deploy it you can simply run (it's the default command specified in the `docker-compose.yml`): 
+
+    docker-compose run deployer
+
+This will print a simple word count to the output console, you can view it by checking the logs of the taskmanager as follows:
+
+    docker-compose logs -f taskmanager
```
Comment (mrooding) [inline README.md]: Would be better to surround all these commands by `` so that they're formatted

File: README.md [hunk 6]
```diff
@@ -16,6 +16,37 @@ Currently, it supports several features:
...
+
+Finally if you want to update the job, you can run:
+
+    docker-compose run deployer update -job-name-base "Windowed WordCount" -file-name "/tmp/flink-sample-job/flink-stateful-wordcount-assembly-0.jar" -run-args "-p 2 -d -c WordCountStateful" -jar-args "--intervalMs 1000" -savepoint-dir "/data/flink/savepoints"
```
Comment (mrooding) [inline README.md]: Maybe split this one into multiple lines for readability?

File: README.md (hunk 5 above)
Comment (nielsdenissen) [reply README.md]: when indented with a tab like here it puts it in a code block automatically

**Pull request merged**